pkg/x/http/trace.go

```diff
@@ -0,0 +1,519 @@
+	dump, err := httputil.DumpRequestOut(reqClone, true)
+	if err != nil {
+		return "", err
+	}
+
+	return string(dump), nil
+}
+
+// dumpResponse dumps and redacts sensitive information from the response
+func (tt *traceTransport) dumpResponse(resp *http.Response) (string, error) {
+	// Read response body
+	var bodyBytes []byte
+	if resp.Body != nil {
+		var err error
+		bodyBytes, err = io.ReadAll(resp.Body)
+		if err != nil {
+			return "", err
+		}
+		// Restore original body
+		resp.Body = io.NopCloser(bytes.NewReader(bodyBytes))
+	}
+
+	// Clone response for redaction
+	respClone := &http.Response{
+		Status:           resp.Status,
+		StatusCode:       resp.StatusCode,
+		Proto:            resp.Proto,
+		ProtoMajor:       resp.ProtoMajor,
+		ProtoMinor:       resp.ProtoMinor,
+		Header:           resp.Header.Clone(),
+		ContentLength:    resp.ContentLength,
+		TransferEncoding: resp.TransferEncoding,
+		Close:            resp.Close,
+		Uncompressed:     resp.Uncompressed,
+		Trailer:          resp.Trailer,
+		Request:          resp.Request,
+	}
+
+	// Redact sensitive headers
+	redactHeaders(respClone.Header)
+
+	// Set redacted body
+	if len(bodyBytes) > 0 {
+		redactedBody := tt.redactBody(bodyBytes, resp.Header.Get("Content-Type"))
+		respClone.Body = io.NopCloser(bytes.NewReader(redactedBody))
+		respClone.ContentLength = int64(len(redactedBody))
+	}
+
+	// Dump the redacted response
+	dump, err := httputil.DumpResponse(respClone, true)
+	if err != nil {
+		return "", err
+	}
+
+	return string(dump), nil
+}
+
+// redactHeaders redacts sensitive headers
+func redactHeaders(headers http.Header) {
+	for _, header := range sensitiveHeaders {
+		for k := range headers {
+			if strings.EqualFold(k, header) {
+				headers[k] = []string{redactedColor(redactedText)}
+			}
+		}
+	}
+}
+
+// redactQueryParams redacts sensitive query parameters
+func redactQueryParams(u *url.URL) *url.URL {
+	// Clone URL to avoid modifying the original
+	cloned, _ := url.Parse(u.String())
+
+	values := cloned.Query()
+	for _, param := range sensitiveQueryParams {
+		for k := range values {
+			if strings.EqualFold(k, param) {
+				values[k] = []string{redactedColor(redactedText)}
+			}
+		}
+	}
+	cloned.RawQuery = values.Encode()
+
+	return cloned
+}
+
+// redactBody redacts sensitive information from request/response bodies using various methods including Trivy's secret scanner
+func (tt *traceTransport) redactBody(body []byte, contentType string) []byte {
+	// Check if body is too large
+	if len(body) > maxBodySize {
+		return []byte(fmt.Sprintf("<body too large: %d bytes>", len(body)))
+	}
+
+	// Check if body is binary
+	if isBinaryContent(contentType) || isBinaryData(body) {
+		return []byte(redactedColor(binaryRedactText))
+	}
+
+	// Start with the original content
+	redacted := string(body)
+
+	// First, use Trivy's secret scanner for detection
+	scanResult := tt.secretScanner.Scan(secret.ScanArgs{
+		FilePath: "http-request.txt",
```

Comment:
nit.
```suggestion
		FilePath: "http-body.txt",
```
We redact boies for both require and response, so this name can be a bit confusing.
but the code will work with "http-request.txt" too.
so it's up to you to decide whether it's worth changing

Reply:
A file name doesn't matter as we don't use it, although the file extension does. I accepted your change.

Reply:
But we check filePath to skip by allow built-in rules
e.g. we can't use `http-test.txt`

But both names are valid :)